### PR TITLE
script:  Remove magic numbers

### PR DESCRIPTION
**Pull request by dajohi**
This adds two new constants, MAX_OPS_PER_SCRIPT and
MAX_PUBKEYS_PER_MULTISIG.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Concept ACK.

**Comment by jamesob** [comment]
utACK

**Comment by jonasschnelli** [comment]
utACK

**Comment by morcos** [comment]
NACK, there is little reward here for the inherent risk in changing consensus code.  This is even a compiler warning for comparing signed and unsigned.

**Comment by paveljanik** [comment]
This is a diff between master and this PR build:

```
+script/interpreter.cpp:276:46: warning: comparison of integers of different signs: 'int' and 'const unsigned int' [-Wsign-compare]
+            if (opcode > OP_16 && ++nOpCount > MAX_OPS_PER_SCRIPT)
+                                  ~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~
+script/interpreter.cpp:872:54: warning: comparison of integers of different signs: 'int' and 'const unsigned int' [-Wsign-compare]
+                    if (nKeysCount < 0 || nKeysCount > MAX_PUBKEYS_PER_MULTISIG)
+                                          ~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~
+script/interpreter.cpp:875:34: warning: comparison of integers of different signs: 'int' and 'const unsigned int' [-Wsign-compare]
+                    if (nOpCount > MAX_OPS_PER_SCRIPT)
+                        ~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~
+3 warnings generated.
```
So NACK.

Why don't you use `#define` instead? This was you could *prove* it is OK by comparing binaries.

**Comment by dcousens** [comment]
concept/utACK after the rebase.

**Comment by sipa** [comment]
utACK

**Comment by paveljanik** [comment]
ACK

File: src/script/script.h [hunk 1]
```diff
@@ -17,7 +17,14 @@
 #include <string>
 #include <vector>
 
-static const unsigned int MAX_SCRIPT_ELEMENT_SIZE = 520; // bytes
+// Maximum number of non-push operations per script
+#define MAX_OPS_PER_SCRIPT 201
```
Comment (laanwj) [inline src/script/script.h]: Please don't use defines to declare constants, follow the rest of the code and use `static const <type>`

Comment (dcousens) [inline src/script/script.h]: They look like `int` from a quick glance.

**Comment by dajohi** [comment]
updated to use `static const int`

**Comment by paveljanik** [comment]
ACK

**Comment by dcousens** [comment]
ACK

**Comment by fanquake** [comment]
utACK

**Pull request merged**